server/src/main/java/org/elasticsearch/index/analysis/AnalyzerComponents.java

```diff
@@ -49,14 +50,11 @@
         this.analysisMode = mode;
     }
 
-    static AnalyzerComponents createComponents(String name, Settings analyzerSettings, final Map<String, TokenizerFactory> tokenizers,
+    static AnalyzerComponents createComponents(String name, Settings analyzerSettings, final Function<String, TokenizerFactory> tokenizers,
             final Map<String, CharFilterFactory> charFilters, final Map<String, TokenFilterFactory> tokenFilters) {
         String tokenizerName = analyzerSettings.get("tokenizer");
-        if (tokenizerName == null) {
-            throw new IllegalArgumentException("Custom Analyzer [" + name + "] must be configured with a tokenizer");
```

Comment:
Do we make sure the name is set elsewhere or is this not necessary anymore? Otherwise it might be good to keep, I haven'r checked what the follow-up exception would be if the name is "null" and the function is called. Might be more cryptic.

Reply:
The check has been moved to `AnalysisRegistry#buildMapping`, line 414